neo/rawio/intanrawio.py

```diff
@@ -407,14 +424,19 @@ def _get_analogsignal_chunk_header_attached(self, i_start, i_stop, stream_index,
             sl0 = i_start % block_size
             sl1 = sl0 + (i_stop - i_start)
 
-        # raw_data is a structured memmap with a field for each channel_id
-        for chunk_index, channel_id in enumerate(channel_ids):
-            data_chan = self._raw_data[channel_id]
-            if multiple_samples_per_block:
-                sigs_chunk[:, chunk_index] = data_chan[block_start:block_stop].flatten()[sl0:sl1]
-            else:
-                sigs_chunk[:, chunk_index] = data_chan[i_start:i_stop]
+        # For all streams raw_data is a structured memmap with a field for each channel_id
+        if not stream_is_digital:
+            sigs_chunk = np.zeros((i_stop - i_start, len(channel_ids)), dtype=dtype)
 
+            for chunk_index, channel_id in enumerate(channel_ids):
+                data_chan = self._raw_data[channel_id]
+                if multiple_samples_per_block:
+                    sigs_chunk[:, chunk_index] = data_chan[block_start:block_stop].flatten()[sl0:sl1]
+                else:
+                    sigs_chunk[:, chunk_index] = data_chan[i_start:i_stop]
+        else:  # For digital data the channels come interleaved in a single field so we need to demultiplex
+            digital_raw_data = self._raw_data[field_name].flatten()
```

Comment:
![image](https://github.com/NeuralEnsemble/python-neo/assets/6429509/bdbde3f0-9138-4871-855e-398c56fecca1)

When you read the numpy structured arrays you read blocks the blocks as columns.  So you need to reshape this as if you were concatenating all the blocks.

Reply:
I understand the flattening, but why aren't we using the 

`self._raw_data[field_name].flatten()[sl0:sl1]` for the header attached?

Reply:
I think I was lazy to implement this because I thought the memory footprint is small? Would you like to give it a go?

We would need to pass this:

```
            digital_raw_data = self._raw_data[field_name][block_start:block_stop].flatten()[sl0:sl1]
```

And then demuliplex only that piece.

Reply:
I see what you mean. So you always load the full and then take what you want. I don't think it is necessary. This vector should always be relatively small compared to the amplifier data, so I don't think we need to worry about memory footprint. We can do that later if needed...